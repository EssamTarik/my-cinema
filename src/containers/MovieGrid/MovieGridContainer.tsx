import React from 'react';
import { IProps } from './interfaces';
import { MovieGrid } from '../../components';

const MovieGridContainer = ({ movies, favorites, addFavorite, removeFavorite }: IProps) => (
  <MovieGrid
    favorites={favorites}
    onAddToFavorites={addFavorite}
    onRemoveFromFavorites={removeFavorite}
    movies={movies}
  />
);

export default MovieGridContainer;