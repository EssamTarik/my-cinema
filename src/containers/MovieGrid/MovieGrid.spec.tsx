jest.mock('../../components', () => ({ MovieGrid: () => <div className='movie-grid' /> }))

import React from "react";
import { render, cleanup } from "@testing-library/react";

const movieMock = jest.requireMock('movie');

import MovieGridContainer from "./MovieGridContainer";

describe('MovieGridContainer', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <MovieGridContainer movies={[movieMock]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
})

