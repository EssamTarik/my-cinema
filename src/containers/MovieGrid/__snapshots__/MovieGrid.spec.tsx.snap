// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGridContainer renders 1`] = `
<DocumentFragment>
  <div
    class="movie-grid"
  />
</DocumentFragment>
`;
